modules/ROOT/pages/patterns/shortest-paths.adoc

```diff
@@ -0,0 +1,537 @@
+
+The previous section showed how to apply a filter before the shortest path selection by the use of parentheses.
+Placing a path variable declaration before the shortest path keywords, however, places it outside the scope of the parentheses.
+To reference a path variable in a pre-filter, it has to be declared inside the parentheses.
+
+To illustrate, consider this example that returns all shortest paths from `Hartlebury` to each of the other `Stations`:
+
+.Query
+[source,cypher]
+----
+MATCH p = SHORTEST 1 (:Station {name: 'Hartlebury'})--+(b:Station)
+RETURN b.name AS destination, length(p) AS pathLength
+ORDER BY pathLength, destination
+----
+
+.Result
+[role="queryresult",options="header,footer",cols="2*m"]
+|===
+| destination | pathLength
+
+| "Droitwich Spa" | 1
+| "Bromsgrove" | 2
+| "Worcester Foregate Street" | 2
+| "Worcester Shrub Hill" | 2
+| "Ashchurch" | 3
+| "Cheltenham Spa" | 3
+| "Worcestershire Parkway Ll" | 3
+| "Pershore" | 4
+
+2+d|Rows:8
+
+|===
+
+If the query is altered to only include routes that have an even number of stops, adding a `WHERE` clause at the `MATCH` level will not work, because it would be a post-filter.
+It would return the results of the previous query with all routes with an odd number of stops removed:
+
+.Query
+[source,cypher]
+----
+MATCH p = SHORTEST 1 (:Station {name: 'Hartlebury'})--+(b:Station)
+WHERE length(p) % 2 = 0
+RETURN b.name AS destination, length(p) AS pathLength
+ORDER BY pathLength, destination
+----
+
+.Result
+[role="queryresult",options="header,footer",cols="2*m"]
+|===
+| destination | pathLength
+
+| "Bromsgrove" | 2
+| "Worcester Foregate Street" | 2
+| "Worcester Shrub Hill" | 2
+| "Pershore" | 4
+
+2+d|Rows: 4
+|===
+
+To move the predicate to a pre-filter, the path variable should be referenced from within the parentheses, and the shortest routes with an even number of stops will be returned for all the destinations:
+
+.Query
+[source,cypher]
+----
+MATCH SHORTEST 1
+  (p = (:Station {name: 'Hartlebury'})--+(b:Station) 
+    WHERE length(p) % 2 = 0 )
+RETURN b.name AS destination, length(p) AS pathLength
+ORDER BY pathLength, destination
+----
+
+.Result
+[role="queryresult",options="header,footer",cols="2*m"]
+|===
+| destination | pathLength
+
+| "Bromsgrove" | 2
+| "Worcester Foregate Street" | 2
+| "Worcester Shrub Hill" | 2
+| "Ashchurch" | 4
+| "Cheltenham Spa" | 4
+| "Droitwich Spa" | 4
+| "Pershore" | 4
+| "Worcestershire Parkway Ll" | 4
+
+2+d|Rows:8
+|===
+
+[[shortest-path-graph-patterns]]
+== Single shortest path per graph pattern
+
+xref:patterns/non-linear-patterns.adoc#graph-patterns[Graph patterns] can contain more than one path pattern.
+The exception is when one of those path patterns has a selective shortest path keyword (For more information, see xref:patterns/reference.adoc#shortest-paths-rules-selective[Syntax and semantics -> selective path selectors]), in which case it can be the only path pattern in the graph pattern.
```

Comment:
```suggestion
The exception is when one of those path patterns has a selective shortest path keyword (for more information, see xref:patterns/reference.adoc#shortest-paths-rules-selective[Syntax and semantics -> selective path selectors]), in which case it can be the only path pattern in the graph pattern.
```